internal/db/nosql/cassandra/option_test.go

```diff
@@ -19,1026 +19,842 @@ package cassandra
-		// Uncomment this line if the option returns an error, otherwise delete it
-		// err error
+		err error
 	}
 	type test struct {
-		name string
-		args args
-		want want
-		// Use the first line if the option returns an error. otherwise use the second line
-		// checkFunc  func(want, *T, error) error
-		// checkFunc  func(want, *T) error
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, *T, error) error
 		beforeFunc func(args)
 		afterFunc  func(args)
 	}
+	defaultCheckFunc := func(w want, obj *T, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got error = %v, want %v", err, w.err)
+		}
 
-	// Uncomment this block if the option returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T, err error) error {
-	       if !errors.Is(err, w.err) {
-	           return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
-	       }
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", obj, w.obj)
-	       }
-	       return nil
-	   }
-	*/
-
-	// Uncomment this block if the option do not returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T) error {
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", obj, w.c)
-	       }
-	       return nil
-	   }
-	*/
+		if diff := comparator.Diff(obj, w.obj, clientComparatorOpts...); diff != "" {
+			return errors.New(diff)
+		}
+		return nil
+	}
 
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           hosts: nil,
-		       },
-		       want: want {
-		           obj: new(T),
-		       },
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           hosts: nil,
-		           },
-		           want: want {
-		               obj: new(T),
-		           },
-		       }
-		   }(),
-		*/
+		{
+			name: "set host success",
+			args: args{
+				hosts: []string{"vald.vdaas.org"},
+			},
+			want: want{
+				obj: &T{
+					hosts: []string{"vald.vdaas.org"},
+				},
+			},
+		},
+		{
+			name: "set host success if hosts is nil",
+			args: args{
+				hosts: nil,
+			},
+			want: want{
+				obj: &T{
+					hosts: nil,
+				},
+				err: func() error {
+					var nilStr []string
+					return errors.NewErrInvalidOption("hosts", nilStr)
+				}(),
+			},
+		},
+		{
+			name: "set host twice success",
+			args: args{
+				hosts: []string{"vald.vdaas.org"},
+			},
+			want: want{
+				obj: &T{
+					hosts: []string{"vald.vdaas.org"},
+				},
+			},
+			checkFunc: func(w want, obj *T, err error) error {
+				if err := defaultCheckFunc(w, obj, err); err != nil {
+					return err
+				}
+
+				// set host to the same obj again
+				if err := WithHosts("hosts1")(obj); err != nil {
+					return err
+				}
```

Comment:
how about set host by beforeFunc?

Reply:
In this test case i want to test the host is appended after the second call. 
I think the current implementation is correct. what do you think?

Reply:
it is correct at the view point of whether work well or not.
but, this pattern is the first time, and I think we can use beforeFunc() instead of your implementation.
I'd like to know which one is better

Reply:
You are right 🤔

Reply:
fixed in this commit.
https://github.com/vdaas/vald/pull/644/commits/5cf14cda999ef13cbe46f5f5d8b97dd70885a900
thank you